src/App.tsx

```diff
@@ -15,48 +16,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  const isAlphabeticSort = sortType === SortType.ALPHABET;
+  const isLengthSort = sortType === SortType.LENGTH;
+
+  if (isAlphabeticSort) {
+    visibleGoods.sort((g1, g2) => g1.localeCompare(g2));
+  }
+
+  if (isLengthSort) {
+    visibleGoods.sort((g1, g2) => g1.length - g2.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const isResetButtonShown = sortType !== SortType.NONE || isReversed;
+  const isAlphabeticSort = sortType === SortType.ALPHABET;
+  const isLengthSort = sortType === SortType.LENGTH;
+
+  const goods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const resetHandler = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const reverseHandler = () => {
+    setIsReversed((currentReversed) => !currentReversed);
+  };
+
+  const sortHandler = (type: SortType) => {
+    setSortType(type);
+  };
```

Comment:
Not really right way to name handlers. Check [checklist](https://github.com/mate-academy/react_list-of-goods-hooks/blob/master/checklist.md) pt.4